sql/2021/pwa/install_events.sql

```diff
@@ -0,0 +1,85 @@
+#standardSQL
+# SW install events
+
+CREATE TEMPORARY FUNCTION getInstallEvents(windowEventListeners ARRAY<STRING>, windowPropertiesInfo ARRAY<STRING>)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+  return [...new Set([...windowEventListeners ,...windowPropertiesInfo])];
+} catch (e) {
+  return [e];
```

Comment:
Is it not good to return these? So if a large proportion of pages return debug output instead of real data then that suggests you JavaScript might be fishy?

if only a small proportion return this then unlikely to impact results.

The concern with NOT returning this is errors like that get hidden.

Reply:
It's ok return debugging info while you're still refining the query, similar to using the sample_data tables, but for the "official" queries they should only return the actual data.

Reply:
Well hopefully the exceptions should never be triggered if the custom metric and UDF are written correctly so it’s hopefully a moot point!

But still concerned with a full query run throwing some error that wasn’t seen in sample_data and it being hidden if this isn’t returned.

Reply:
By all means, include debug output while you're iterating and revise the query as needed. When you're done though, remove the debug output before checking the query in and saving the results to Sheets.

Reply:
Or an error message instead of silently swallowing the error and pretending nothing's wrong.

But fine. They are gone.